tools/src/main/scala/scala/scalanative/optimizer/pass/Canonicalization.scala

```diff
@@ -0,0 +1,52 @@
+package scala.scalanative
+package optimizer
+package pass
+
+import analysis.ClassHierarchy._
+import nir._, Inst.Let
+
+/** Moves constant operands in Op.Bin and Op.Comp to the right. */
+class Canonicalization extends Pass {
+
+  override def preInst = {
+    case Let(n, Op.Bin(bin, ty, lhs, rhs))
+        if (commutativeBin(bin) && constant(lhs)) =>
+      Seq(Let(n, Op.Bin(bin, ty, rhs, lhs)))
+
+    case Let(n, Op.Comp(comp, ty, lhs, rhs))
+        if (commutativeComp(comp) && constant(lhs)) =>
+      Seq(Let(n, Op.Comp(comp, ty, rhs, lhs)))
+  }
+
+  private def commutativeBin(bin: Bin): Boolean = {
+    import Bin._
+    bin match {
+      case Iadd | Fadd | Imul | Fmul | And | Or | Xor => true
```

Comment:
Yeah, for the other passes, I was having a hard time knowing what could be a problem, and what couldn't. Should this be true for constant folding as well ? (coming next)